sage.py

```diff
@@ -215,259 +171,271 @@ def parse(unparsed_data, alert_labels=[], slim=False):
     for (i,h) in _hosts:
         if h not in hosts.keys():
             hosts[h] = 0 if len(hosts.values())==0 else max(hosts.values())+1'''
-    
-    #print(cats)
-    #print(len(cats))
-    #print(data[0][1], data[0][3])
-    #print(data[1][1], data[1][3])
-    print('Reading # alerts: ', len(data))
-    
+
+    print('Reading # alerts: ', len(parsed_data))
+
     if slim:
-        print(len(data), len(alert_labels))
+        print(len(parsed_data), len(alert_labels))
         j = 0
-        for i,al in enumerate(alert_labels):
+        for i, al in enumerate(alert_labels):
             spl = al.split(',')
             source = spl[0]
             dest = spl[1]
             mcat = int(spl[-1][:-1])
             cat = spl[2]
-            
+
             if source == badIP or dest == badIP or cat == 'Not Suspicious Traffic':
                 continue
             if spl[2] == 'Attempted Information Leak' and FILTER:
                 continue
-                
```

Comment:
A bit confusing looking at this diff (because of the weird matching of blank lines)... Essentially, I have moved the plotting part from `remove_duplicates` method into a separate method. Btw, these plots look pretty nice :)